supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActorSpec.scala

```diff
@@ -2,28 +2,40 @@ package cromwell.backend.impl.tes
 
 import common.mock.MockSugar
 import cromwell.backend.BackendJobDescriptorKey
-import cromwell.backend.BackendSpec.buildWdlWorkflowDescriptor
+import cromwell.backend.BackendSpec.{buildWdlWorkflowDescriptor, whenReady}
+import cromwell.backend.async.PendingExecutionHandle
+import cromwell.backend.standard.{StandardAsyncExecutionActorParams, StandardAsyncJob}
+import cromwell.core.TestKitSuite
 import cromwell.core.logging.JobLogger
 import cromwell.core.path.{DefaultPathBuilder, NioPath}
 import cromwell.filesystems.blob.{BlobFileSystemManager, BlobPath, WSMBlobSasTokenGenerator}
 import cromwell.filesystems.http.HttpPathBuilder
 import org.mockito.ArgumentMatchers.any
-import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.flatspec.AnyFlatSpecLike
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.prop.TableDrivenPropertyChecks
 import wom.graph.CommandCallNode
 
 import java.time.Duration
 import java.time.temporal.ChronoUnit
+import scala.collection.immutable.Map
+import scala.concurrent.{ExecutionContextExecutor, Future}
 import scala.util.{Failure, Try}
 
 class TesAsyncBackendJobExecutionActorSpec
-    extends AnyFlatSpec
+    extends TestKitSuite
+    with AnyFlatSpecLike
     with Matchers
     with MockSugar
     with TableDrivenPropertyChecks {
   behavior of "TesAsyncBackendJobExecutionActor"
```

Comment:
I think we would really benefit from some tests of the polling behavior, since that seems to be the bulk of the new logic in this branch. I think it's important to know what web requests are made and what status is returned when calling `pollStatusAsync` in different conditions. We could try to make that function pure and test it directly in unit tests, or do some work to spin up our own mock `TesAsyncBackendJobExecutionActor` to use for unit tests. 

I think it would also be great to have a test that makes sure we're emitting the expected metadata when calling `onTaskComplete` in different conditions.  

Happy to help!